testkit/src/api.rs

```diff
@@ -280,4 +279,97 @@ impl TestKitApi {
             false,
         )
     }
+
+    /// Converts iron Response to ApiError.
+    ///
+    /// # Panics
+    ///
+    /// - Panics if the response has a non-error response status.
+    fn response_to_api_error(response: Response) -> ApiError {
+        let status = response.status.expect("Status header is not set");
+
+        fn error(response: Response) -> String {
+            let body = response::extract_body_to_string(response);
+
+            fn match_error(body: &str) -> Option<String> {
```

Comment:
Nitpick: I think this function could be moved one level up (i.e., at the same level as `error`) and named something like `description` or `extract_description`.